subsystem.py

```diff
@@ -0,0 +1,94 @@
+robot_path = 'src/main/java/frc/robot/'
+subsystem_class = '''package frc.robot.subsystems;
+
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+
+import frc.robot.maps.subsystems.${CLASSNAME}Map;
+
+public class ${CLASSNAME} extends SmartSubsystemBase {
+
+    private ${CLASSNAME}Map map;
+
+    public ${CLASSNAME}(${CLASSNAME}Map map) {
+        this.map = map;
+    } 
+
+    @Override
+    public void reset() {
+        // Nothing to reset here
+    }
+
+    @Override
+    public void periodic() {
+        // This method will be called once per scheduler run
+        // Use this for any background processing
+    }
+
+    @Override
+    public void simulationPeriodic() {
+        // This method will be called once per scheduler run during simulation
+    }
+
+    @Override
+    public void safeState() {
+
+    }
+}'''
+
+subsystem_map = '''package frc.robot.maps.subsystems;
+
+public class ${CLASSNAME}Map {
+    public ${CLASSNAME}Map() {
+
+    }
+}
+'''
+
+def insert_into(filename: str, keyword: str, line:str, before:bool = False):
+    with open(f'{robot_path}{filename}','r') as fp:
+        contents = fp.read()
+    if before:
+        new_contents = contents.replace( keyword, f'\n{line}\n{keyword}'  )
+    else:
+        new_contents = contents.replace(keyword, f'{keyword}\n{line}\n')
+    with open(f'{robot_path}{filename}','w') as fp:
+        fp.write(new_contents)
+
+def save_template(filename, template, **kwargs):
+    new_template = template
+    for key, value in kwargs.items():
+        new_template = new_template.replace(f'${{{key}}}',value)
+
+    with open(f'{robot_path}{filename}','w') as fp:
+        fp.write(new_template)
+
+class_name = input('Subsystem Class Name: ')
+
+# Make sure that it is a proper class name
+class_name = ''.join( list(map(lambda s: s[0].upper() + s[1:], class_name.split())) )
```

Comment:
Clever solution, but we can simplify it a bit:

```py
class_name = ''.join(name.title() for name in class_name.split())
```

Reply:
Ugh wait I just realized that wouldn't work for acronyms... That's annoying.

Reply:
Yeah the idea here is to allow for the inputting of an actual class name, or a string of words and it would produce the same output. List comprehension does make it cleaner though I just forgot how to use it.